doc/srv6/srv6-hld-v19.md

```diff
@@ -0,0 +1,552 @@
+
+# About this Manual
+
+This document provides general information about the Segmentation Routing over IPv6 feature implementation in SONiC. It is based on IETF RFC 8754 and RFC 8986.
+
+# 1 Introuduction and Scope
+
+This document describes the Functionality and High level design of the SRv6 feature.
+
+SRv6 has been widely adopted as an IPv6 based SDN solution, which provides programming ability, TE capabilities,  and deployment simplicity to network administrators.  With current support from a rich ecosystem, including major ASIC manufactures,  networking vendors and open source communities, the deployment of SRv6 is accelerating. We want to add SRv6 into SONIC to benefit users in DC as well as beyond DC.  
+
+The following are some use cases for SRv6 deployment:
+
+- v4/6VPN, EVPN over best-effort
+- Traffic steering over TE policy 
+
+In SRv6 domain,  TE policy associated with SID list could be configured on headend nodes, to steer traffic with SRH encapsulation. When traffic reaches egress nodes,  the packets are processed based on local defined functions, for example SID list decapsulation and FIB lookup in a particular VRF . 
+
+# 2 Feature Requirements
+
+## 1.1 Functional Requirements
+
+This section describes the SONiC requirements for SRv6 feature in phases:
+
+At a high level the following should be supported:
+
+Phase #1
+
+​	Should be able to perform the role of SRv6 domain headend node, and endpoint node, more specific:
+
+- Support H.Encaps, H.Encaps.Red, END.B6.Encaps, END.B6.Encaps.Red
+- Support END, END.X
+- Support END.DT46
+- Support traffic steering on SID list
+- Support anycast SID
+
+Later phases:
+
+- Support uSID/G-SID
+- Other programming functions
+- Support HMAC option
+- Support sBFD for SRv6 
+
+This document will focus on Phase #1, while keep the design extendable for future development
+
+## 1.2 Configuration and Management Requirements
+
+1. User should be able to enable SRv6 globally or per interface
+
+2. User should be able to configure SID list for encapsulation
+
+3. User should be able to configure SRv6 steering policy
+
+4. User should be able to configure endpoint action and corresponding argument for matched local SID
+
+## 1.3 Scalability Requirements
+
+- Maximum number of end.DT46 SID: 2k
+- Maximum number  of Binding SID: 1k
+- Maximum number of SR policy: 1k
+- Maximum SID-list per SR Policy: 4
+- Maximum number of SID-list: 4k
+- SRH depth with regular SID: 3
+
+## 1.4 Warm Boot Requirements
+
+Warm reboot is intended to be supported for planned system, swss and BGP warm reboot.
+
+ 
+
+# 2 Feature Design
+
+![draw-configdb](images/drawing-configdb-frr3.png)
+
+Before FRR is ready, we will use static configuration to set SIDs and apply policy for TE. It enables basic SRv6 operation and populates SRv6 into ASIC, allows SRv6 data plan forwarding. More complicated SRv6 policy can be enabled when SRv6 is fully supported in FRR and passed from FRR to fpmsyncd.
```

Comment:
data plane - typo

Reply:
fixed.

Reply:
@goldenrye , its not addressed

Reply:
@caizhenghui-juniper Thanks for reviewing the HLD. For phase 1, the controller is supposed to avoid route conflicts with FRR. This can be achieved by using different VRFs or prefixes between routes from controller or FRR. For phase 2, all routes update need to go through FRR, who will take care of the route conflicts.